### DISCUSSION TITLE
Equivalent of show rule in template function?

### DISCUSSION THREAD

**Discussion by Holt59** [typst/typst#774] [Q&A]
I'd like to provide a template that people can use by simply doing

```typ
#import "template.typ": conf

#show: doc => conf(doc)  // with extra options
```

In this template, I would like to customize headings and raw output, which I can do in the main file with

```
#show heading: set text(size: 12pt)

#show heading.where(level: 1): (name) => {
  return upper(name)
}

#show raw: set text(font: "Cascadia Code PL")
```

Is there a way to "include" these show-rules in the `conf()` function? I've tried including them by simply removing the `#` but it did not work.

**Comment by Kaligule** [comment]
The Typst Examples Book [describes](https://sitandr.github.io/typst-examples-book/book/basics/tutorial/templates.html) your problem

> Because set and show rules are only active in their current scope, they will not affect content in a file you imported your file into.

...and how to solve it:

In the template

```typ
#let apply-template(body) = [
    #show heading: set text(size: 12pt)
    #show heading.where(level: 1): (name) => {
        return upper(name)
    }
    #show raw: set text(font: "Cascadia Code PL")
    
    #body
]
```

and in the main file:

```typ
#import "template.typ": apply-template
#show: apply-template
```
